### PR TITLE
Fix 1.0.0 issues

### PR DESCRIPTION
**Pull request by radoering**
Fix issues encountered during review of python-poetry/poetry#5413

`test_exporter_doesnt_confuse_repeated_packages` in `test_exporter.py` requires an upstream fix.

Depends-on: python-poetry/poetry#5435

### CODE REVIEW

File: src/poetry_plugin_export/exporter.py [hunk 1]
```diff
@@ -98,17 +98,10 @@ def _export_requirements_txt(self, cwd: Path, output: IO | str) -> None:
         ops = solver.solve().calculate_operations()
         packages = sorted((op.package for op in ops), key=lambda pkg: pkg.name)
 
-        # Get project dependencies.
-        if self._groups is not None:
```
Comment (abn) [inline src/poetry_plugin_export/exporter.py:102]: I do not think this change is correct. The intended behaviour should be if groups are specified, export them. And if no groups are specified, default to `poetry export --with=default` to emulate current behavriour. I tink the  `self._poetry.package.without_optional_dependency_groups()` should be dropped and we should use `list(self._groups or ["default"])` instead.

The `dev=True` is superfluous, as we should really remove that option in poetry as well.

File: src/poetry_plugin_export/exporter.py [hunk 2]
```diff
@@ -161,11 +154,9 @@ def _export_requirements_txt(self, cwd: Path, output: IO | str) -> None:
 
                     hashes.append(f"{algorithm}:{h}")
 
-                if hashes:
-                    line += " \\\n"
-                    for i, h in enumerate(hashes):
-                        suffix = " \\\n" if i < len(hashes) - 1 else ""
-                        line += f"    --hash={h}{suffix}"
+                for h in hashes:
+                    line += f" \\\n    --hash={h}"
```
Comment (abn) [inline src/poetry_plugin_export/exporter.py:158]: Why was this changed? This was an upstream fix in the in-tree export command iirc..

File: src/poetry_plugin_export/exporter.py (hunk 1 above)
Comment (radoering) [reply src/poetry_plugin_export/exporter.py:102]: If I haven't overlooked something, removing these lines is fine because they are duplicates (see L81-L86). As for the default behavior, I'll check if I can apply your proposal in the lines that are still existing.

At the moment the `dev=True` is necessary because otherwise depdendencies not in the default group are always dropped and tests fail. (Nevertheless, I'll check again after applying the proposed changes.)

File: src/poetry_plugin_export/exporter.py (hunk 2 above)
Comment (radoering) [reply src/poetry_plugin_export/exporter.py:158]: Isn't the that equivalent?

Before change:
If there are hashes a newline is appended first. Then, each hash is appended and after each hash a newline. Except for the last hash: For the last hash no new line is appended.

After the change:
For each hash a newline is appended first and afterwards the hash.

Maybe I'm blind right now, but I don't see any difference.

File: src/poetry_plugin_export/exporter.py (hunk 1 above)
Comment (radoering) [reply src/poetry_plugin_export/exporter.py:102]: I restored the default behavior of `Exporter.export()` and verified that `dev=True` is required to get correct results because dev defaults to `False`.

However, the default behavior of `Exporter.export()` is not as relevant as I thought first because `groups` are always set by the command.

File: src/poetry_plugin_export/exporter.py (hunk 2 above)
Comment (abn) [reply src/poetry_plugin_export/exporter.py:155]: You are correct, my bad. I was tired when I looked at the code last, and it reminded me of another merge conflict I had looked at before.

File: src/poetry_plugin_export/exporter.py (hunk 1 above)
Comment (abn) [reply src/poetry_plugin_export/exporter.py:102]: Groups LGTM now.

Yes, the `dev=True` is required. It is fairly useless with group, I wonder if it is easier to just not add the parameter here, and update the `use-export-plugin` branch upstream to switch locker to default to `dev=True` behaviour? This will save us a deprecation cycle. Since the 1.0.0 release of the plugin requires the upstream branch thanks to https://github.com/python-poetry/poetry-plugin-export/blob/main/pyproject.toml#L19.

Comment (abn) [reply src/poetry_plugin_export/exporter.py:102]: https://github.com/python-poetry/poetry/pull/5413/commits/642775bf9c7ad2ac74e9495fd8290525bb9060c9

File: src/poetry_plugin_export/exporter.py [hunk 3]
```diff
@@ -98,17 +95,10 @@ def _export_requirements_txt(self, cwd: Path, output: IO | str) -> None:
...
-            )
-        else:
-            root_package = self._poetry.package.without_optional_dependency_groups()
-
         for dependency_package in self._poetry.locker.get_project_dependency_packages(
-            project_requires=root_package.all_requires,
-            project_python_marker=root_package.python_marker,
+            project_requires=root.all_requires,
+            project_python_marker=root.python_marker,
+            dev=True,
```
Comment (abn) [inline src/poetry_plugin_export/exporter.py:101]:
```suggestion
```

See: https://github.com/python-poetry/poetry/pull/5413/commits/642775bf9c7ad2ac74e9495fd8290525bb9060c9

**Review by abn — CHANGES_REQUESTED** [review]

File: src/poetry_plugin_export/exporter.py (hunk 3 above)
Comment (abn) [reply src/poetry_plugin_export/exporter.py]: Committed suggestion to trigger the CI run.

File: poetry.lock [hunk 4]
```diff
@@ -372,7 +373,7 @@ virtualenv = "(>=20.4.3,<20.4.5 || >=20.4.7)"
 type = "git"
 url = "https://github.com/abn/poetry.git"
 reference = "use-export-plugin"
-resolved_reference = "c5f74f605f0eecd5aedcfb75d06e5a7230ab1b16"
+resolved_reference = "aee5e054cb9978dc4908a8715a8c85a98a209b58"
```
Comment (abn) [inline poetry.lock:376]:
```suggestion
resolved_reference = "642775bf9c7ad2ac74e9495fd8290525bb9060c9"
```

File: src/poetry_plugin_export/exporter.py [hunk 5]
```diff
@@ -78,12 +78,9 @@ def _export_requirements_txt(self, cwd: Path, output: IO | str) -> None:
...
-            root = self._poetry.package.with_dependency_groups(
-                list(self._groups), only=True
-            )
-        else:
-            root = self._poetry.package.without_optional_dependency_groups()
+        root = self._poetry.package.with_dependency_groups(
+            list(self._groups or ["default"]), only=True
+        )
 
         locked_repository = self._poetry.locker.locked_repository(True)
```
Comment (abn) [inline src/poetry_plugin_export/exporter.py:85]:
```suggestion
        locked_repository = self._poetry.locker.locked_repository()
```

**Review by abn — APPROVED** [review]
